src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs

```diff
@@ -380,9 +381,49 @@ private static long ParseTimeValue(JToken jToken, string claimName)
         /// <param name="validationParameters">A <see cref="TokenValidationParameters"/>  required for validation.</param>
         /// <returns>Returns a <see cref="SecurityKey"/> to use for signature validation.</returns>
         /// <remarks>If key fails to resolve, then null is returned</remarks>
-        internal static SecurityKey ResolveTokenSigningKey(string kid, string x5t, TokenValidationParameters validationParameters)
+        private static SecurityKey ResolveTokenSigningKeyWithConfig(string kid, string x5t, TokenValidationParameters validationParameters)
```

Comment:
Check SigningKeys at the beginning and return null, no need to continue:

```
if (validationParameters.Configuration?.SigningKeys == null)
{
    return null;
}
```

Reply:
Good suggestion; just added this in. :)

Reply:
The checks, line 391 & 405, can be removed.

Reply:
Whoops, good catch. Fixed.